File: src/main/js/structure/MagicNumberSection.js
function MagicNumberSection(bytesArray) {

    this.getMagicNumber = function () {
        return magicNumber;
    }

    this.getStartByteIndex = function() {
        return 0;
    }

    this.length = function() {
        return 4;
    }

    this.getEndByteIndex = function() {
        return this.getStartByteIndex() + this.length();
    }

    this.contains = function (index) {
        if(index >= this.getStartByteIndex() && index < this.getStartByteIndex() + this.length()) {
            return true;
        }
        else {
            return false;
        }
    }

    this.getCssClass = function() {
        return "magic-number-section";
    }

    this.getBytecode = function (index) {
        return "Magic Number" + "<br/>"
        + formatAsOneByteHexString(0) + " " + formatAsOneByteHexString(bytesArray[0]) + "<br/>"
        + formatAsOneByteHexString(1) + " " + formatAsOneByteHexString(bytesArray[1]) + "<br/>"
        + formatAsOneByteHexString(2) + " " + formatAsOneByteHexString(bytesArray[2]) + "<br/>"
        + formatAsOneByteHexString(3) + " " + formatAsOneByteHexString(bytesArray[3]) + "<br/>" ;
    }

    var magicNumber = getFourBytes(bytesArray, this.getStartByteIndex());
}

File: src/main/java/com/vgrazi/bytecodeexplorer/structure/MagicNumberSection.java
package com.vgrazi.bytecodeexplorer.structure;

import com.vgrazi.bytecodeexplorer.ui.swing.PointSelector;
import com.vgrazi.bytecodeexplorer.utils.Utils;

/**
 * Created by vgrazi on 8/13/15.
 */
public class MagicNumberSection implements ClassFileSection {
    private int magicNumber;
    private byte[] bytes;

    public MagicNumberSection(byte[] bytes) {
        this.bytes = bytes;
        magicNumber = Utils.getFourBytes(bytes, getStartByteIndex());
    }

    public long getMagicNumber() {
        return magicNumber;
    }

    @Override
    public int length() {
        return 4;
    }

    @Override
    public int getStartByteIndex() {
        return 0;
    }

    public String toString() {
        int startByte = getStartByteIndex();
        return Utils.formatAsFourByteHexString(startByte) + " Magic number: " +
            "<span style='" + PointSelector.getStyleForByte(startByte, 1)     + "'>" + Utils.formatAsOneByteHexString(Utils.getOneByte(bytes, startByte)) + "</span>" +
            "<span style='" + PointSelector.getStyleForByte(startByte + 1, 1) + "'>" + Utils.formatAsOneByteHexString(Utils.getOneByte(bytes, startByte + 1)) + "</span>" +
            "<span style='" + PointSelector.getStyleForByte(startByte + 2, 1) + "'>" + Utils.formatAsOneByteHexString(Utils.getOneByte(bytes, startByte + 2)) + "</span>" +
            "<span style='" + PointSelector.getStyleForByte(startByte + 3, 1) + "'>" + Utils.formatAsOneByteHexString(Utils.getOneByte(bytes, startByte + 3)) + "</span> = " + magicNumber;
    }
}
